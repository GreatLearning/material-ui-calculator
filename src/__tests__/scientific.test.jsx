import renderer from 'react-test-renderer';
import React from 'react';
import Scientific from '../scientific';

it('renders correctly', () => {
  const tree = renderer.create(<Scientific onInput={jest.fn()} />).toJSON();
  expect(tree).toMatchSnapshot();
});
