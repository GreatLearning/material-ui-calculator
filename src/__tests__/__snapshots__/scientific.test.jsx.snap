// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Scientific-scientific-1"
>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "(",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": ")",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "Ï€",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "log",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "sin",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "cos",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "tan",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "ln",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "asin",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "acos",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "atan",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "n!",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "âˆš",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "xÂ²",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "xÂ³",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "x<sup>y</sup>",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "1/x",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "e",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
  <div
    className="Pad-pad-3 Scientific-pad-2"
    style={undefined}
  >
    <button
      className="MuiButtonBase-root-11 MuiIconButton-root-5 Pad-button-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="-1"
      type="button"
    >
      <span
        className="MuiIconButton-label-10"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "abs",
            }
          }
        />
      </span>
      <span
        className="MuiTouchRipple-root-14"
      />
    </button>
  </div>
</div>
`;
